src/platform/options/default_configuration.cpp

```diff
@@ -303,7 +305,12 @@ void mo::DefaultConfiguration::parse_arguments(
         for (auto const& token : unparsed_arguments)
             tokens.push_back(token.c_str());
         if (!tokens.empty()) unparsed_arguments_handler(tokens.size(), tokens.data());
-
+        if (options.is_set("mir-version"))
+		{
+			std::ostringstream the_version_of_mir;
+			the_version_of_mir << "MIR Platform version " << MIR_VERSION_MAJOR << "." << MIR_VERSION_MINOR << "." << MIR_VERSION_MICRO << "." << std::endl;
+			BOOST_THROW_EXCEPTION(mir::ExitWithOutput(the_version_of_mir.str()));
```

Comment:
This shouldn't be throwing an exception, a clean exit is appropriate when the ask is to show the version string.

Reply:
Updated to just print to stdout and exit cleanly. It seemed too simple to be true, but I can't think of a better way and it seems to shut down cleanly.